.github/actions/cleanup/project.go

```diff
@@ -0,0 +1,59 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+func PrepareListToDelete(projects *mongodbatlas.Projects, deleteAll bool, lifetimeInHours int) ([]*mongodbatlas.Project, error) {
+	var projectList []*mongodbatlas.Project
+	if deleteAll {
+		projectList = projects.Results
+	} else {
+		for _, project := range projects.Results {
+			createdTime, err := time.Parse(time.RFC3339, project.Created) // check format
+			if err != nil {
+				return nil, fmt.Errorf("error parsing project creation time: %w", err)
+			}
+			if time.Since(createdTime) < time.Duration(lifetimeInHours)*time.Hour {
+				projectList = append(projectList, project)
+			}
+		}
+	}
+	return projectList, nil
+}
+
+func DeleteProjects(ctx context.Context, client mongodbatlas.Client, projectList []*mongodbatlas.Project) bool {
```

Comment:
Are other resources(alerts, custom roles...) deleted with the project or will we need to update the "cleaner" always we add new features?

Reply:
Only a small amount of entities may prevent the project from deleting. At the moment, the cleaner deletes everything that the operator supports and can prevent the project from being deleted.